examples/hwc_example.cpp

```diff
@@ -81,47 +81,51 @@ int main(int argc, char* argv[])
     videoPipe = dpipe_lookup("video-0"))
     */
 
-    //create hwc receiver
-    VirtualHwcReceiver* receiver = new VirtualHwcReceiver(cfg, cmd_handler);
-    IOResult ior = receiver->start();
-    int index = 0;
-    while (!std::get<0>(ior)) {
-        index ++;
-
-        if (index % 300 == 0) {
-            printf("tag: index(%d)\n", index);
+    try {
+        //create hwc receiver
+        VirtualHwcReceiver* receiver = new VirtualHwcReceiver(cfg, cmd_handler);
+        IOResult ior = receiver->start();
+        int index = 0;
+        while (!std::get<0>(ior)) {
+            index ++;
+
+            if (index % 300 == 0) {
+                printf("tag: index(%d)\n", index);
+            }
+
+            if (index == 3600) {
+                printf("tag: index, (%s)!!\n", "setmode");
+                IOResult ior = receiver->setMode(1920, 1080);
+            int res = std::get<0>(ior);
+            if (res<0) {
+                printf("tag: setMode(%s)\n", "failed");
+            } else {
+                printf("tag: setMode(%s)\n", "successfully");
+            }
+            }
+
+            if (index == 5000) {
+                IOResult ior = receiver->stop();
+                int res = std::get<0>(ior);
+                if (res<0) {
+                    printf("tag: stop(%s)\n", "failed");
+                } else {
+                    printf("tag: stop(%s)\n", "successfully");
+                }
+            }
+
+            if (index == 6000) {
+                delete receiver;
+                receiver = NULL;
```

Comment:
Can you add a break here to make sure that you won't go to the next iteration which will crash in calling `receiver->SetMode`?